instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ProcessResource.java

```diff
@@ -35,6 +34,9 @@ public final class ProcessResource {
   // Important: This is statically used in buildResource, so must be declared/initialized first.
   private static final Pattern JAR_FILE_PATTERN =
       Pattern.compile("^\\S+\\.(jar|war)", Pattern.CASE_INSENSITIVE);
+  // scrub values for system properties containing "secret" or "password" in the name
+  private static final Pattern SCRUB_PATTERN =
+      Pattern.compile("(-D.*(password|secret).*=).*", Pattern.CASE_INSENSITIVE);
```

Comment:
Would it be worth to also scrub properties containing `pwd` or `credential`?

Reply:
I'll leave that to @trask to decide.

Reply:
IMO, it's another option to provide an environment variable for users so that they can set some keys themselves.

Reply:
I think ok to go with just `password` and `secret` initially and see if we get further reports